.github/workflows/ci.yml

```
@@ -120,12 +121,12 @@ jobs:
   frontend-tests:
     runs-on: ubuntu-20.04
     steps:
-    - uses: actions/checkout@v3
+    - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4
 
     - name: Setup NodeJS
-      uses: actions/setup-node@v3
+      uses: actions/setup-node@60edb5dd545a775178f52524783378180af0d1f8 # v4
```

```suggestion
      uses: actions/setup-node@v4.0.2
```

Reference: https://github.com/actions/setup-node/releases/tag/v4.0.2